python/python/lance/dataset.py

```diff
@@ -106,6 +106,9 @@ def list_indices(self) -> List[Dict[str, Any]]:
         return self._ds.load_indices()
 
     def index_statistics(self, index_name: str) -> Dict[str, Any]:
+        """
+        deprecated: use .stats.index_stats() instead
+        """
         return json.loads(self._ds.index_statistics(index_name))
```

Comment:
Also should emit a deprecation warning, so users see it. Like here:

https://github.com/lancedb/lance/blob/0f2de196a123a0a506eb19f5fa4ebd705ca9a256/python/python/lance/dataset.py#L873-L877

Reply:
Thanks good suggestion! made this change